examples_tests/01.HelloWorld/main.cpp

```diff
@@ -286,13 +302,236 @@ int main()
+		renderPassBeginInfo.renderArea.offset = { 0, 0 };
+		renderPassBeginInfo.renderArea.extent = swapchainExtent;
+		renderPassBeginInfo.clearValueCount = 1u;
+		renderPassBeginInfo.pClearValues = &clearColor;
+
+		vkCmdBeginRenderPass(commandBuffers[i], &renderPassBeginInfo, VK_SUBPASS_CONTENTS_INLINE);
+
+		// Do nothing
+
+		vkCmdEndRenderPass(commandBuffers[i]);
+
+		assert(vkEndCommandBuffer(commandBuffers[i]) == VK_SUCCESS);
+	}
+
+	video::ISwapchain* rawPointerToSwapchain = swapchain.get();
+
+	uint32_t currentFrameIndex = 0u;
+	while (!windowShouldClose_Global)
+	{
+		video::IGPUSemaphore* acquireSemaphore_frame = acquireSemaphores[currentFrameIndex].get();
+		video::IGPUSemaphore* releaseSemaphore_frame = releaseSemaphores[currentFrameIndex].get();
+		video::IGPUFence* fence_frame = frameFences[currentFrameIndex].get();
+
+		assert(device->waitForFences(1u, &fence_frame, true, ~0ull) == video::IGPUFence::ES_SUCCESS);
+
+		uint32_t imageIndex;
+		swapchain->acquireNextImage(~0ull, acquireSemaphores[currentFrameIndex].get(), nullptr,
+			&imageIndex);
+
+		// At this stage the final color values are output from the pipeline
+		// Todo(achal): Not really sure why are waiting at this pipeline stage for
+		// acquiring the image to render
+		VkPipelineStageFlags pipelineStageFlags = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+
+		VkSubmitInfo submitInfo = { VK_STRUCTURE_TYPE_SUBMIT_INFO };
+		submitInfo.waitSemaphoreCount = 1u;
+		submitInfo.pWaitSemaphores = &vk_acquireSemaphores[currentFrameIndex];
+		submitInfo.pWaitDstStageMask = &pipelineStageFlags;
+		submitInfo.commandBufferCount = 1u;
+		submitInfo.pCommandBuffers = &commandBuffers[imageIndex];
+		submitInfo.signalSemaphoreCount = 1u;
+		submitInfo.pSignalSemaphores = &vk_releaseSemaphores[currentFrameIndex];
+
+		// Make sure you unsignal the fence before expecting vkQueueSubmit to signal it
+		// once it finishes its execution
+		device->resetFences(1u, &fence_frame);
+
+		VkResult result = vkQueueSubmit(vk_graphicsQueue, 1u, &submitInfo, vk_frameFences[currentFrameIndex]);
+		assert(result == VK_SUCCESS);
+
+		// asset::E_PIPELINE_STAGE_FLAGS waitDstStageFlags = asset::E_PIPELINE_STAGE_FLAGS::EPSF_COLOR_ATTACHMENT_OUTPUT_BIT;
+
+		// video::IGPUQueue::SSubmitInfo submitInfo = {};
+		// submitInfo.waitSemaphoreCount = 1u;
+		// submitInfo.pWaitSemaphores = &acquireSemaphore_frame;
+		// submitInfo.pWaitDstStageMask = &waitDstStageFlags;
+		// submitInfo.signalSemaphoreCount = 1u;
+		// submitInfo.pSignalSemaphores = &releaseSemaphore_frame;
+		// submitInfo.commandBufferCount = 1u;
+		// submitInfo.commandBuffers = ;
+		// assert(graphicsQueue->submit(1u, &submitInfo, fence_frame));
+
+		video::IGPUQueue::SPresentInfo presentInfo;
+		presentInfo.waitSemaphoreCount = 1u;
+		presentInfo.waitSemaphores = &releaseSemaphore_frame;
+		presentInfo.swapchainCount = 1u;
+		presentInfo.swapchains = &rawPointerToSwapchain;
+		presentInfo.imgIndices = &imageIndex;

```

Comment:
where are the wait dest stages?

Reply:
seems you should wait in the present stage (if such exists) ?

Reply:
What?

There is no wait dest param.

Reply:
ah yeah on the present there's none 🤦 

the image acquire semaphore on the submit should wait for the color attachment write stage though, but thats unrelated to present I guess.